lib/battle/CObstacleInstance.cpp

```diff
@@ -142,6 +142,19 @@ void SpellCreatedObstacle::battleTurnPassed()
 		turnsRemaining--;
 }
 
+std::vector<BattleHex> BattlefieldLimiterObstacle::getAffectedTiles() const
+{
+	std::vector<BattleHex> impassableTiles;

```

Comment:
This method does not make sense. You should make BattlefieldLimiterObstacle 1-tile and put your obstacles to positions during battle setup.